[Merged by Bors] - Describe new delegation strategy (#4562)
bevyengine/bevy: CONTRIBUTING.md

Some Bevy Org members are also Triage Team members. These people can label and close issues and PRs but do not have merge rights or any special authority within the community. Existing Org members can vist the Bevy Engine Org page and navigate to the Triage Team to request membership. Once again, if you are interested don't hesitate to apply. We generally accept membership requests.

Merge rights within the org are relatively centralized: this requires a large amount of trust when it comes to ethics, technical ability, and ability to enforce consistent project direction.

The current structure is as follows:

* @cart is our project lead, and has final say on controversial decisions
* There is a small group of other maintainers (@alice-i-cecile, @mockersf and @superdump), who have merge rights but abide by the following rules:
  * Trivial PRs can be merged without approvals

>   * Relatively uncontroversial PRs can be merged following approval from at least one other community member with appropriate expertise.

Reviewer: I'd personally prefer if this said "at least two community approvals", I don't like that my vote has that much power even if it's for something uncontroversial. If it's something important then multiple people will approve it, otherwise it's probably not a priority to many people.

Author: ```suggestion
  * Relatively uncontroversial PRs can be merged following approval from at least two other community members with appropriate expertise.
```

  * Controversial PRs are added to a backlog for @cart to address once two maintainers agree that they are ready.

>   * If 45 days elapse without action on a controversial PR (approval, feedback or an explicity request to defer), they can be merged without project lead approval.

Reviewer: ```suggestion
  * If 45 days elapse without action on a controversial PR (approval, feedback or an explicit request to defer), they can be merged without project lead approval.
```

* The Bevy org is made up of trusted community contributors: this is a relatively low bar, and org members help triage and maintain the project.
* Community contributors (this means you!) can freely open issues, submit PRs and review PRs to improve Bevy.
  * As discussed above, community reviews on PRs are incredibly helpful to enable maintainers to merge in uncontroversial PRs in a timely fashion.

### Classifying PRs

This strategy relies on a classification of PRs into three categories: **trivial**, **uncontroversial** and **controversial**.

> When making PRs, try to split out more controversial changes from less controversial ones, in order to make your work easier to review and merge.

Reviewer 2: We should probably add a line here calling out the `S-Controversial` label explicitly.

Author: ```suggestion
When making PRs, try to split out more controversial changes from less controversial ones, in order to make your work easier to review and merge.
PRs that are deemed controversial will receive the `S-Controversial` label, and will have to go through the more thorough review process.
```

PRs are trivial if there is no reasonable argument against them. This might include:

* fixing dead links
* removing dead code or dependencies
* typo and grammar fixes

PRs are controversial if there is serious design discussion required, or a large impact to contributors or users. Factors that increase controversy include:

1. Changes to project-wide workflow or style.
2. New architecture for a large feature.
3. PRs where a serious tradeoff must be made.
4. Heavy user impact.
5. New ways for users to make mistakes (footguns).
6. Introductions of `unsafe` code.
7. Large-scale code reorganization.
8. High levels of technical complexity.
9. Adding a dependency.

> Finally, by the law of the excluded middle, changes are relatively uncontroversial if they are neither trivial or controversial.

Reviewer 2: "by the law of excluded middle" is a niche enough concept that I'm not sure its particularly helpful here. I think removing that text improves clarity. We might also want to put "relatively uncontroversial" in quotes whenever it shows up because we're defining our own specific meaning for that phrase.

Author: ```suggestion
Finally, changes are "relatively uncontroversial" if they are neither trivial or controversial.
```
